redis/redis.go

```diff
@@ -80,9 +83,12 @@ func GetRedisClient(conf *RedisConfig) (redis.UniversalClient, error) {
 	var rc redis.UniversalClient
 	var tlsConfig *tls.Config
 
-	if conf.UseTLS {
-		tlsConfig = &tls.Config{
-			MinVersion: tls.VersionTLS12,
+	if (conf.TLS != nil && conf.TLS.Enabled) || conf.UseTLS {
+		var err error
+		tlsConfig, err = conf.TLS.ClientTLSConfig()
```

Comment:
conf.TLS could be nil here and would panic.

Reply:
https://go.dev/play/p/YhBSjgK_yDY
@davidzhao

Reply:
your playground is demonstrating the check around if.. that's not where the panic happens. It takes place within `ClientTLSConfig` where you access member variables of a nil TL struct.

Reply:
@davidzhao Yes, you're right!
Fixed